src/main/java/com/stripe/model/Charge.java

```diff
@@ -493,4 +493,104 @@ public Charge update(Map<String, Object> params, RequestOptions options) throws
         options);
   }
   // </editor-fold>
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public static class AlternateStatementDescriptors extends StripeObject {
+    String kana;
+    String kanji;
+  }
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public static class FraudDetails extends StripeObject {
+    public static final String USER_REPORT = "user_report";
+
+    String userReport;
+    String stripeReport;
+  }
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public static class Level3 extends StripeObject {
+    String customerReference;
+    List<LineItem> lineItems;
+    String merchantReference;
+    String shippingAddressZip;
+    String shippingFromZip;
+    Long shippingAmount;
+
+    @Getter
+    @Setter
+    @EqualsAndHashCode(callSuper = false)
+    public static class LineItem extends StripeObject {
+      Long discountAmount;
+      String productCode;
+      String productDescription;
+      Long quantity;
+      Long taxAmount;
+      Long unitCost;
+    }
+  }
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public static class Outcome extends ApiResource {
+    String networkStatus;
+    String reason;
+    String riskLevel;
+    @Getter(AccessLevel.NONE) @Setter(AccessLevel.NONE) ExpandableField<Rule> rule;
+    String sellerMessage;
+    String type;
+
+    // <editor-fold desc="rule">
+    /**
+     * Returns the {@code rule} object, if expanded. If not expanded, use {@link #getRuleId()} to
+     * get the ID.
+     *
+     * @return the {@code rule} ID
+     * @deprecated In recent API versions, this attribute is no longer automatically expanded.
+     *     Prefer using the {@link #getRuleId()} and {@link #getRuleObject()} methods instead.
+     * @see <a href="https://stripe.com/docs/upgrades#2017-02-14">API version 2017-02-14</a>
+     */
+    @Deprecated
+    public Rule getRule() {
+      return (this.rule != null) ? this.rule.getExpanded() : null;
+    }
+
+    @Deprecated
+    public void setRule(Rule rule) {
+      this.rule = new ExpandableField<Rule>(rule.getId(), rule);
+    }
+
+    public String getRuleId() {
+      return (this.rule != null) ? this.rule.getId() : null;
+    }
+
+    public void setRuleId(String ruleId) {
+      this.rule = setExpandableFieldId(ruleId, this.rule);
+    }
+
+    public Rule getRuleObject() {
+      return (this.rule != null) ? this.rule.getExpanded() : null;
+    }
+
+    public void setRuleObject(Rule rule) {
+      this.rule = new ExpandableField<Rule>(rule.getId(), rule);
+    }
+    // </editor-fold>
+
+    @Getter
+    @Setter
+    @EqualsAndHashCode(callSuper = false)
+    public static class Rule extends StripeObject implements HasId {
```

Comment:
Note that this class is doubly nested: `ChargeOutcomeRule` is now `Charge.Outcome.Rule`.